### PR TITLE
🗺️ Planes

### PR DESCRIPTION
**Pull request by Quentintnrl**
Add planes primitives

### CODE REVIEW

File: src/Transform.cpp [hunk 1]
```diff
@@ -7,6 +7,7 @@
 */
 
 #include "Transform.hpp"
+#include <iostream>
```
Comment (RenardFute) [inline src/Transform.cpp:10]: ?

File: src/primitives/Properties/PlaneProperties.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+void PlaneProperties::setHeight(float height)
+{
+    _height = height;
+}
+
+void PlaneProperties::setWidth(float width)
+{
+    _width = width;
+}
```
Comment (RenardFute) [inline src/primitives/Properties/PlaneProperties.cpp:57]:
```suggestion
}

```

File: src/primitives/Properties/PlaneProperties.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/*
+** EPITECH PROJECT, 2024
+** RTX
+** File description:
+** No file there , just an epitech header example .
+** You can even have multiple lines if you want !
+*/
+
+#include "primitives/Properties/PlaneProperties.hpp"
+#include <iostream>
```
Comment (RenardFute) [inline src/primitives/Properties/PlaneProperties.cpp:10]: unused include

File: src/primitives/Plane.cpp [hunk 4]
```diff
@@ -0,0 +1,62 @@
+/*
+** EPITECH PROJECT, 2024
+** RTX
+** File description:
+** No file there , just an epitech header example .
+** You can even have multiple lines if you want !
+*/
+
+#include "primitives/Plane.hpp"
+#include <iostream>
```
Comment (RenardFute) [inline src/primitives/Plane.cpp:10]: unused include

File: src/main.cpp
Comment (RenardFute) [inline src/main.cpp:1]: For the love of god, please stop rewriting the main class each time you add a primitive, just create a function somewhere that fill the factories and give them back to you. Each of your pull request seems to modify 15 lines here where 2 should be necessary, and it's just prone to miss reading on our side, which can cause line deletions or creation that would have been unwanted.

**Review by RenardFute — CHANGES_REQUESTED** [review]
In the overall it's ok, but I would really like you to change the main function one time for all and move all the factories stuff somewhere else.
Also, can you remove your stdio includes ?

**Review by aimericdrk — APPROVED** [review]

**Review by RenardFute — APPROVED** [review]

**Review by RenardFute — APPROVED** [review]

**Review by RenardFute — APPROVED** [review]

**Pull request merged**